### PR TITLE
Add Lightning Sounds + Configurable options for them

### PR DESCRIPTION
**Pull request by greenwolf5**
## Additions
* Adds LightningCharge sound effect, with a method playLightningbendingChargingSound located in FireAbility, that gets called in Lightning when the move is charging. Allows config support following the similar structure of LightningSound
* Adds LightningHit sound effect, with a method playLightningbendingHitSound located in FireAbility, that gets called in Lightning when the move hits an entity. Allows config support following the similar structure of LightningSound
* Adds playLightningbendingSound() when the lightning is progressing

## Misc changes
* updated the pom.xml to https since it was found maven doesn't support the previous method

### CODE REVIEW

**Comment by ChristopherWMM** [comment]
Thank you for your contributions, we really appreciate it! That being said, there are a few things we were hoping you could change to make this PR something we would consider merging.

1. Right now, you are including some old commits due to your branch being out of date with `wip`. Ideally, your PR should only include the new commits you are making.
2. There are some miscellaneous comments that are not super necessary. We would prefer that these are removed. Specifically, the following:
    - https://github.com/ProjectKorra/ProjectKorra/pull/1161/files#diff-618fee85c0577a7dd0a34f0f8c56e9f4e84729c5de56a4650bcf80df75eac8c3R210-R211
    - https://github.com/ProjectKorra/ProjectKorra/pull/1161/files#diff-7b451201bde2696c679a9996f0759b16b113eff19cd6ea33935cd3eb51d7e691R215
    - https://github.com/ProjectKorra/ProjectKorra/pull/1161/files#diff-7b451201bde2696c679a9996f0759b16b113eff19cd6ea33935cd3eb51d7e691R251
3. If you are going to add documentation comments to methods, I would prefer that they be standard javadocs.
4. In Lightning.java, you create a new `Random` object for each instance. It would be better to use `Math.random()` or better yet [ThreadLocalRandom](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/util/concurrent/ThreadLocalRandom.html) to prevent having to make new Random objects. If this is new, check out [this post](https://www.baeldung.com/java-thread-local-random#threadlocalrandom-example) for more info.

Please feel free to reach out if you have any questions or need help making the above changes!

File: src/com/projectkorra/projectkorra/ability/FireAbility.java [hunk 1]
```diff
@@ -203,6 +203,38 @@ public static void playLightningbendingSound(final Location loc) {
...
+		if (getConfig().getBoolean("Properties.Fire.PlaySound")) {
+			final float volume = (float) getConfig().getDouble("Properties.Fire.LightningCharge.Volume");
+			final float pitch = (float) getConfig().getDouble("Properties.Fire.LightningCharge.Pitch");
+
+			Sound sound = Sound.BLOCK_BEEHIVE_WORK;
+			try {
+				sound = Sound.valueOf(getConfig().getString("Properties.Fire.LightningCharge.Sound"));
+			} catch (final IllegalArgumentException exception) {
+				ProjectKorra.log.warning("Your current value for 'Properties.Fire.LightningCharge.Sound' is not valid.");
+			} finally{
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/ability/FireAbility.java:217]: You should have a space after `finally`.

```java
} finally {
```

File: src/com/projectkorra/projectkorra/ability/FireAbility.java [hunk 2]
```diff
@@ -203,6 +203,38 @@ public static void playLightningbendingSound(final Location loc) {
...
+		if (getConfig().getBoolean("Properties.Fire.PlaySound")) {
+			final float volume = (float) getConfig().getDouble("Properties.Fire.LightningHit.Volume");
+			final float pitch = (float) getConfig().getDouble("Properties.Fire.LightningHit.Pitch");
+
+			Sound sound = Sound.ENTITY_LIGHTNING_BOLT_THUNDER;
+			try {
+				sound = Sound.valueOf(getConfig().getString("Properties.Fire.LightningHit.Sound"));
+			} catch (final IllegalArgumentException exception) {
+				ProjectKorra.log.warning("Your current value for 'Properties.Fire.LightningHit.Sound' is not valid.");
+			}	finally {
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/ability/FireAbility.java:233]: There should be a space between the closing brace of the catch and finally, not a tab.

```java
} finally {
```

File: src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java [hunk 3]
```diff
@@ -204,6 +207,10 @@ public void progress() {
 					final Location loc = this.player.getEyeLocation().add(this.player.getEyeLocation().getDirection().normalize().multiply(1.2));
 					loc.add(0, 0.3, 0);
 					playLightningbendingParticle(loc, 0.2F, 0.2F, 0.2F);
+					if(ThreadLocalRandom.current().nextDouble() < .2){
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java:210]: There should be a space between the if and opening parenthesis. Additionally, there should be a space between the closing parenthesis and the opening brace.

```java
if (ThreadLocalRandom.current().nextDouble() < .2) {
```

File: src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java [hunk 4]
```diff
@@ -139,8 +141,9 @@ public Lightning(final Player player) {
 	public void electrocute(final LivingEntity lent) {
 		playLightningbendingSound(lent.getLocation());
 		playLightningbendingSound(this.player.getLocation());
+		playLightningbendingHitSound(lent.getLocation());
+		playLightningbendingHitSound(this.player.getLocation());
 		DamageHandler.damageEntity(lent, this.damage, this);
-
 		if (Math.random() <= this.stunChance) {
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java:147]: Might as well change this and other instances of `Math.random()` to use `ThreadLocalRandom.current().nextDouble()`

File: src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java [hunk 5]
```diff
@@ -235,6 +242,9 @@ public void progress() {
 				final Location localLocation2 = new Location(this.player.getWorld(), d7, newY, d8);
 				playLightningbendingParticle(localLocation2);
 				this.particleRotation += 1.0D / d3;
+				if(ThreadLocalRandom.current().nextDouble() < .2){
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java:245]: There should be a space between the if and opening parenthesis. Additionally, there should be a space between the closing parenthesis and the opening brace.

```java
if (ThreadLocalRandom.current().nextDouble() < .2) {
```

File: src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java [hunk 6]
```diff
@@ -499,6 +509,10 @@ public void run() {
 			if (this.count > 5) {
 				this.cancel();
 			} else if (this.count == 1) {
+				if(ThreadLocalRandom.current().nextDouble() < .1){
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/lightning/Lightning.java:512]: There should be a space between the if and opening parenthesis. Additionally, there should be a space between the closing parenthesis and the opening brace.

```java
if (ThreadLocalRandom.current().nextDouble() < .1) {
```

**Review by ChristopherWMM — CHANGES_REQUESTED** [review]

**Review by ChristopherWMM — APPROVED** [review]
Great job! Thank you for working out all those requested details! 👍

**Review by Simplicitee — APPROVED** [review]

**Pull request merged**